tools/mo/openvino/tools/mo/utils/version.py

```diff
@@ -75,3 +73,15 @@ def get_simplified_ie_version(env=dict(), version=None):
     if m and len(m.groups()) == 3:
         return simplify_version(m.group(3))
     return simplify_version(version)
+
+
+def extract_version_hash(full_version: str):
+    """
+    extracts 11 character hash from the full version string
```

Comment:
why do you extract just 11 characters? Please extract the full hash for the comparison.

Reply:
I extract only 11 characters because even now in some circumstances we have 11 character hash, e.g.
**from wheel in nightyly:**
OpenVINO runtime version:    2022.1.0-6311-**a90bb1ff017**
Model Optimizer version:        2022.1.0-6311-**a90bb1ff017**

**from pip:**
Inference Engine version:       2021.4.2-3976-**0943ed67223**-refs/pull/539/head
Model Optimizer version:        2021.4.2-3976-**0943ed67223**-refs/pull/539/head

Moreover in the future we will be always returned in unified format and only 11 characters will be awailable
![image](https://user-images.githubusercontent.com/5703039/153170737-2e0e6030-8510-44ce-8463-dde169c203fa.png)

Reply:
For your example above `2022.1.0-6311-a90bb1f` this parse function returns None. Why should we rely on 11 character hash, if git uses SHA-1 that counts 40 characters in hash? And what is confidence that this length will not change in the future?

Reply:
Thanks for noticing that from wheel in nightly is shorter. Updated regexp, now 7 character is extracted.

If one of the hashes contains all 40 characters and other is shortened to 7, for comparison I use only firsts 7 characters, so that we consider that versions with hashes are the same
```
a90bb1ff0176c70bda5770dd825a58c5cc620395
a90bb1f
```